freud/environment.pyx

```diff
@@ -512,7 +512,18 @@ cdef class _MatchEnv(_PairCompute):
 
 cdef class EnvironmentCluster(_MatchEnv):
     r"""Clusters particles according to whether their local environments match
-    or not, according to various shape matching metrics.
+    or not, using various shape matching metrics defined in :cite:`Teich2019`.
+
+    Note that two sets of neighbor lists are used: :code:`'env_neighbors'`
+    defines the query particles' environments, and :code:`'neighbors'`
+    defines the neighborhood where query particles' environments are compared
+    to determine the clusters.
+
+    For example, with :code:`'env_neighbors = {num_neighbors: 12}'` and
+    :code:`'neighbors = {'num_neighbors': 15}'`, the :code:`compute` function
+    constructs an environment that contains 12 neighbors for each query
+    particle, and compares it with the environments of the particle's 15 nearest
+    neighbors.
```

Comment:
This should probably go into the docstring of compute rather than the constructor since it's discussing parameters for that function.